session/session_test.go

```diff
@@ -5677,13 +5677,14 @@ func (s *testTiDBAsLibrary) TestMemoryLeak(c *C) {
 	runtime.ReadMemStats(&memStat)
 	oldHeapInUse := memStat.HeapInuse
 
-	for i := 0; i < 10; i++ {
+	for i := 0; i < 20; i++ {
 		initAndCloseTiDB()
 	}
 
 	runtime.GC()
 	runtime.ReadMemStats(&memStat)
-	c.Assert(memStat.HeapInuse-oldHeapInUse, Less, uint64(150*units.MiB))
+	// before the fix, initAndCloseTiDB for 20 times will cost 900 MB memory, so we test for a quite loose upper bound.
+	c.Assert(memStat.HeapInuse-oldHeapInUse, Less, uint64(300*units.MiB))
```

Comment:
Can you explain how this is expected to work?

I would have thought either `oldHeapInUse` is calculated after initAndCloseTiDB() has been executed 20 times.

Reply:
this unstable test is introduced by https://github.com/pingcap/tidb/pull/28027 , after dom.Close some memory didn't get released.

so repeat some times, leak grows significantly than usual memory. more repeat times gives us a wider bound to assert